spec.md

```diff
@@ -0,0 +1,73 @@
+# Collaborative clock and transport proof-of-concept
+
+This document lays out requirements for a proof-of-concept clock and transport for web audio applications.
+
+## Goals
+
+The goal is to create a simple application where two users collaborate on a shared data structure that syncs over a time-based loop. 
+
+## Definitions
+
+- Server - a backend application that stores and syncs the data structure
+- Client - a frontend application that plays and updates its part of the data structure
+- Shared Data - the shared data structure
+- Local Data - the local version of the data structure. Each client has their own version of Local Data.
+- Clock - a thing that keeps time. Each client uses one to play over the data structure.
+- Loop Start - the time when the loop is started or restarted after loop end
+
+## Constraints
+
+- Local Data and Shared Data use the same schema
+- Clients modify independent parts of the schema. The data structure is shared, but to simplify synchronization, each user can only modify one part of the schema.
+- Each user can only modify their Local Data. They send their Local Data to the Server, and the Server updates Shared Data.
+- Each user has their own Clock. Clocks do not need to stay in sync with each other.
+
+## Clock
+
+Each Client has a Clock that is used to track time across the loop and determine when to restart the loop. A stable clock implementation is outlined in https://blog.paul.cx/post/metronome/.
+
+## Synchronization
+
+Synchronization occurs on the Server and on the Client.
+
+On the Server:
+
+- The Server may receive Local Data from a Client at any time
+- The Server updates the Shared Data with any changes in Local Data
+- The Server sends the updated Shared Data to each Client immediately after the update (should the Server send updates to the Client that requested the update?)
+
+On the Client:
+
+- The Client may receive Shared Data from the Server at any time
+- The Client syncs their Local Data with Shared Data moments before Loop Start
+- The Client sends Local Data updates to the Server immediately after sync but before Loop Start
+
+## Transport
```

Comment:
Can we define "Transport?" The term is generic enough that context is not enough.

Reply:
Yep, will add a definition.